examples/jsm/renderers/webgpu/WebGPURenderPipelines.js

```diff
@@ -799,7 +762,217 @@ const ShaderLib = {
 		void main() {
 			outColor = vec4( 1.0, 0.0, 0.0, 1.0 );
 		}`
+	},
+	// Shader headers for ShaderMaterial
+	shader: {
+		vertexShader: `#version 450
+
+		layout(location = 0) in vec3 position;
+		layout(location = 1) in vec2 uv;
+
+		layout(location = 0) out vec2 vUv;
+
+		layout(set = 0, binding = 0) uniform ModelUniforms {
+			mat4 modelMatrix;
+			mat4 modelViewMatrix;
+		} modelUniforms;
+
+		layout(set = 0, binding = 1) uniform CameraUniforms {
+			mat4 projectionMatrix;
+			mat4 viewMatrix;
+		} cameraUniforms;
+		`,
+		fragmentShader: `#version 450
+
+		layout(location = 0) in vec2 vUv;
+		layout(location = 0) out vec4 outColor;
+		`
 	}
 };
 
+function parseAttributes( vertexShader ) {
+
+	// Find "layout (location = num) in type name" in vertex shader
+	const regex = /^\s*layout\s*\(\s*location\s*=\s*(?<location>[0-9]+)\s*\)\s*in\s+(?<type>\w+)\s+(?<name>\w+)\s*;/gmi;
+	let shaderAttribute = null;
+
+	const attributes = [];
+
+	while ( shaderAttribute = regex.exec( vertexShader ) ) {
+
+		const shaderLocation = parseInt( shaderAttribute.groups.location );
+		const arrayStride = getGLSLArrayStride( shaderAttribute.groups.type );
+		const vertexFormat = getGLSLVertexFormat( shaderAttribute.groups.type );
+
+		attributes.push( {
+			name: shaderAttribute.groups.name,
+			arrayStride: arrayStride,
+			slot: shaderLocation,
+			format: vertexFormat
+		} );
+
+	}
+
+	return attributes.sort( function ( a, b ) {
```

Comment:
Let's do the review step by step^^.

This is very good suggestion which makes the vertex buffer setup more robust. Code like:

```
layout(location = 1) in vec3 position;
layout(location = 0) in vec2 uv;
```
now works too.

I've applied your code in dfe538e053fae75b1d0bfcdd2b353e36b387c5a0. However, it was necessary to use `slot` instead of `shaderLocation` (since it is `undefined`).